Update CONTRIBUTING.md with steps to start local development

nicholas-wu-hs opened the pull request.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/hootsuite/atlantis?pullRequest=169) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/hootsuite/atlantis?pullRequest=169) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/hootsuite/atlantis?pullRequest=169) it.</sub>

anubhavmishra commented:
@nicholas-wu-hs Please do sign the CLA before we can merge it :)

nicholas-wu-hs commented:
Thanks Luke, I've updated the `README.md` and `CONTRIBUTING.md` based on your suggestions in #168 -- you suggested 3 `#`s for setting up atlantis subsections, but I used 2 `#`s to follow the same style in `CONTRIBUTING.md`, let me know if that needs to change.
I've also updated the README ToC to include some other missing sections.

I don't have permissions to close other people's PRs, @nick-hollingsworth-hs please close #168 and sign the CLA when you get a chance :)

lkysow commented on README.md:
```diff
@@ -24,13 +24,23 @@ Read about [Why We Built Atlantis](https://www.atlantis.run/blog/atlantis-releas
 * [Locking](#locking)
 * [Approvals](#approvals)
 * [Production-Ready Deployment](#production-ready-deployment)
+    * [Install Terraform](#install-terraform)
```
let's leave these out of the table of contents. They were left out purposefully so that it was easy to find what people were looking for without reading too many bullets.

lkysow commented on README.md:
```diff
@@ -24,13 +24,23 @@ Read about [Why We Built Atlantis](https://www.atlantis.run/blog/atlantis-releas
+    * [Add GitHub Webhook](#add-github-webhook)
+    * [Create a GitHub Token](#create-a-github-token)
+    * [Start Atlantis](#start-atlantis)
     * [Docker](#docker)
```
the Docker part was left in because that's the one thing in this section that people might be looking for explicitly

lkysow commented on README.md:
```diff
@@ -24,13 +24,23 @@ Read about [Why We Built Atlantis](https://www.atlantis.run/blog/atlantis-releas
+    * [Testing Out Atlantis](#testing-out-atlantis)
 * [Server Configuration](#server-configuration)
 * [AWS Credentials](#aws-credentials)
+    * [Multiple AWS Accounts](#multiple-aws-accounts)
```
leave these headings out for same reason as above

lkysow commented on README.md:
```diff
@@ -446,6 +456,8 @@ However, if you were to lose the data, all you would need to do is run `atlantis
 
 
+## Contributing
+Want to contribute? Check out [CONTRIBUTING](https://github.com/hootsuite/atlantis/blob/master/CONTRIBUTING.md).
```
nice

lkysow commented on CONTRIBUTING.md:
```diff
@@ -1,3 +1,40 @@
+- Create a WebHook in your repo and use the `https` url that `ngrok` printed out after running `ngrok http 4141`. Be sure to append `/events` so your webhook url looks something like `https://efce3bcd.ngrok.io/events`. See [Add GitHub Webhook](https://github.com/hootsuite/atlantis#add-github-webhook).
+- Create a pull request and type `atlantis help`. You should see the request in the `ngrok` and Atlantis logs and you should also see Atlantis comment back.
+
 # Code Style
```
I think code style should be a sub-heading under developing. Let's do that and then keep logging etc under code style (so add a `#` to those)

lkysow commented on CONTRIBUTING.md:
```diff
@@ -1,3 +1,40 @@
+# Developing
+
+## Running Atlantis Locally
+- Get the source code:
```
let's ditch the bullets here since it looks better without them. I had them below because there was multiple steps without a code block to separate them so if the steps had been just separated by newlines it would have been difficult to read.

The heuristic I use is:
* if there's code blocks in between the instructions then don't use bullets
* once there's a case where there's multiple lines without a code block then use bullets

nicholas-wu-hs replied:
Thanks! That sounds good, I was thinking in terms of consistency haha

nicholas-wu-hs commented:
- Removed subheadings in `README.md` ToC except `Docker`. Would we want to keep credits in the ToC? It wasn't there prior to this PR
- Moved whole `Code Style` section under `Developing` in `CONTRIBUTING.md`

lkysow commented:
We can merge after nick signs the CLA.

@anubhavmishra are the e2e tests failing because this is a branch by not us?

anubhavmishra commented:
@lkysow yep, we will need to figure out how we can make it so that is not the case. I think I did most of the work to get e2e to a point where we can make them run without worrying out permissions etc.
